lib/matplotlib/axis.py

```diff
@@ -1413,6 +1419,9 @@ def grid(self, b=None, which='major', **kwargs):
 
                 grid(color='r', linestyle='-', linewidth=2)
         """
+        if b is None and kwargs:
+            b = True  # if any kwargs are supplied, assume you want the grid on
```

Comment:
This must check for `'visible' in kwargs`, or it will break as it does in the tests.